src/twisted/conch/client/knownhosts.py

```diff
@@ -520,19 +529,15 @@ def addHostKey(self, hostname, key):
         self._added.append(entry)
         return entry
 
-    def save(self):
+    def save(self) -> None:
         """
         Save this L{KnownHostsFile} to the path it was loaded from.
         """
         p = self._savePath.parent()
         if not p.isdir():
             p.makedirs()
 
-        if self._clobber:
-            mode = "wb"
-        else:
-            mode = "ab"
-
+        mode: Literal["a", "w"] = "w" if self._clobber else "a"
```

Comment:
Hm... Not sure how the test passed for this code.

I see the file is opend in text mode, but then we write bytes.

----------

```suggestion
        mode: Literal["ab", "wb"] = "wb" if self._clobber else "an"
```

Reply:
`FilePath` always opens files in binary mode; it doesn't have a text mode, so the `b` is superfluous. See the annotation here: https://docs.twistedmatrix.com/en/stable/api/twisted.python.filepath.FilePath.html#open and note that it unconditionally returns `IO[bytes]`.